src/index.ts

```diff
@@ -62,6 +63,43 @@ const PKG = require('../../package.json');
 const v2 = require('./v2');
 const {grpc} = new gax.GrpcClient();
 
+export interface RequestConfig {
+  gaxOpts?: CallOptions;
+  method: string;
+  reqOpts: {};
+  client: string;
+  retryOpts?: {};
+}
+
+export type RequestCallback<T, R = void> = R extends void
+  ? GenericCallback<T>
+  : ResourceCallback<T, R>;
+export interface ResourceCallback<Resource, Response> {
+  (
+    err: ServiceError | null,
+    resource?: Resource | null,
+    response?: Response
+  ): void;
+}
+export interface PagedCallback<Item, Response> {
+  (
+    err: ServiceError | null,
+    results?: Item[] | null,
+    nextQuery?: {} | null,
+    response?: Response | null
+  ): void;
+}
+export interface OptionInterface {
+  gaxOptions?: CallOptions;
+}
+export interface AppProfileOptions extends OptionInterface {
+  routing: Cluster;
+  allowTransactionalWrites?: boolean;
+  description?: string;
+  ignoreWarnings?: string;
+}
+export type CreateAppProfileCallback = GenericCallback<AppProfile>;
+export type CreateAppProfileResponse = [AppProfile];
```

Comment:
If these interfaces are not used in this file, I would move them into `Instance.ts`